lib/addict/routes_helper.ex

```diff
@@ -12,21 +12,40 @@ defmodule Addict.RoutesHelper do
       route_options = options_for_route(route, options[route])
 
       quote do
+        get unquote(route_options[:path]),
+          Addict.AddictController,
+          unquote(route),
+          as: unquote(route),
+          private: %{
+            addict: %{
+              layout: unquote(route_options[:layout]),
+              view: unquote(route_options[:view])
+            }
+          }
+
         post unquote(route_options[:path]),
           unquote(route_options[:controller]),
           unquote(route_options[:action]),
-          as: unquote(route_options[:as])
+          as: unquote(route_options[:as]),
+          private: %{
+            addict: %{
+              layout: unquote(route_options[:layout]),
+              view: unquote(route_options[:view])
+            }
+          }
       end
     end
   end
 
   defp options_for_route(route, options) when is_list(options) do
     path       = route_path(route, options[:path])
-    controller = options[:controller] || Addict.Controller
+    controller = options[:controller] || Addict.AddictController
     action     = options[:action] || route
     as         = route
 
-    %{path: path, controller: controller, action: action, as: as}
+    post_action = "process_#{to_string(action)}" |> String.to_atom
```

Comment:
On the ExampleApp we have to change `UserController.signout/2` to `UserController.process_signout/2`.